frontend/src/utilities/time.ts

```diff
@@ -0,0 +1,23 @@
+export const timeDifference = (current: number, previous: number): string => {
+  const msPerMinute = 60 * 1000;
+  const msPerHour = msPerMinute * 60;
+  const msPerDay = msPerHour * 24;
+  const msPerMonth = msPerDay * 30;
+  const msPerYear = msPerDay * 365;
+
+  const elapsed = current - previous;
+
+  if (elapsed < msPerMinute) {
+    return Math.round(elapsed / 1000) + ' seconds ago';
+  } else if (elapsed < msPerHour) {
+    return Math.round(elapsed / msPerMinute) + ' minutes ago';
+  } else if (elapsed < msPerDay) {
+    return Math.round(elapsed / msPerHour) + ' hours ago';
+  } else if (elapsed < msPerMonth) {
+    return 'approximately ' + Math.round(elapsed / msPerDay) + ' days ago';
+  } else if (elapsed < msPerYear) {
+    return 'approximately ' + Math.round(elapsed / msPerMonth) + ' months ago';
+  } else {
+    return 'approximately ' + Math.round(elapsed / msPerYear) + ' years ago';
```

Comment:
When the image is just imported (in `Importing` phase), the reported `notebokk.uploaded` is `undefined`. See here: https://github.com/opendatahub-io/odh-dashboard/pull/171/files#diff-5128194b7438d4f34d77a69aa1b358cb72d24ed2762f816243148ccf83bb3edeR21

This else branch will be probably hit. For example Openshift console defaults to `Now` in such cases.

Reply:
Updated code to comply with PF guidelines and fixed this issue https://www.patternfly.org/v4/ux-writing/numerics#absolute-or-relative-time